File: #1094/1094.cpp
#include <iostream>
#include <algorithm>
#include <cstring>
#include <cmath>
#include <queue>
#include <stack>

using namespace std;

int x;

void input(){
    cin >> x;
}

void solve(){
    int stick = 64;
    int small = stick;
    int piece = 0;

    while(true){
        if (small > x){
            small /= 2;
        } else {
            x -= small;
            small /= 2;
            piece++;
        }

        if (x == 0) {
            break;
        }
    }

    cout << piece;
}
int main(){
    input();
    solve();
    return 0;
}

File: #1094/1094.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int small = 64;
        int x = in.nextInt();
        int piece = 0;

        while (x != 0) {
            if (small > x){
                small /= 2;
            } else {
                x -= small;
                small /= 2;
                piece++;
            }
        }

        System.out.print(piece);
    }
}

